File: Solutions/C#/Runtime-Error/198-house-robber.cs
/*
Problem Title: 198 - House Robber
Lang: C#
Runtime: N/A
Memory: N/A
Problem Url: https://leetcode.com/problems/house-robber
Submission Url: https://leetcode.com/problems/house-robber/submissions/302027958
*/

public class Solution {
    public int Rob(int[] nums) {
        int length = nums.Length;
        int[,] res = new int[2, length];
        
        for(int index = 0; index < length; index++){
            if(index == 0){
                res[0,index] = 0;
                res[1,index] = nums[index];                
            }
            else{
                res[0,index] = Math.Max(res[0,index-1], res[1,index-1]);
                res[1,index] = res[0,index-1] + nums[index];
            }
        }
        return Math.Max(res[0, length-1], res[1, length-1]);
    }
}

File: Solutions/C++/Wrong-Answer/198-house-robber.cpp
/*
Problem Title: 198 - House Robber
Lang: C++
Runtime: N/A
Memory: N/A
Problem Url: https://leetcode.com/problems/house-robber
Submission Url: https://leetcode.com/problems/house-robber/submissions/593342478
*/

class Solution {
public:
    int rob(vector<int>& nums) {

        if(nums.size() == 1){
            return nums[0];
        }


        vector<vector<int>> dp(2);

        dp[0].push_back(0);
        dp[1].push_back(nums[0]); 

        for(int i = 1; i < nums.size(); i++){

            dp[0].push_back(dp[1][i-1]);
            dp[1].push_back(dp[0][i-1] + nums[i]);

        }

        return max(dp[0][nums.size()-1], dp[1][nums.size()-1]);
    }
};